docs/how_to_auxiliary.rst

```diff
@@ -0,0 +1,243 @@
+How to create an auxiliary
+--------------------------
+
+This tutorial aims to explain the working principle of an Auxiliary by
+providing information on the different Auxiliary interfaces, their purpose
+and the implementation of new auxiliaries.
+
+To provide hints on the implementation of an Auxiliary, an example that
+implements a generic auxiliary is provided, that is not usable but shall
+explain the different concepts and implementation steps.
+
+Different Auxiliary interfaces for different use-cases
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Pykiso provides three different auxiliary interfaces, used as basis for
+any implemented auxiliary. These different interfaces aim to cover every
+possible usage of an auxiliary:
+
+- The :py:class:`~pykiso.interfaces.thread_auxiliary.AuxiliaryInterface`
+  is a `Thread <https://docs.python.org/3.7/library/threading.html#threading.Thread>`_-based auxiliary.
+  It is suited for IO-bound tasks where the reception of data cannot be expected.
+- The :py:class:`~pykiso.interfaces.mp_auxiliary.MpAuxiliaryInterface`
+  is a `Process <https://docs.python.org/3.7/library/multiprocessing.html#multiprocessing.Process>`_-based auxiliary.
+  It is suited for CPU-bound tasks where the reception of data cannot be expected and its processing can be CPU-intensive.

```

Comment:
Can it be extended with the information that it utilizes all the cores available on the CPU?

Reply:
Done